### ISSUE TITLE
[@typescript-eslint/no-unused-expressions] False positive with ternaries

### ISSUE THREAD

**Issue by astorije** [typescript-eslint/typescript-eslint#1793]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-expressions": "error"
  }
}
```

```tsx
const condition = true;
const foo = 1;
const bar = 2;
const onClick = (id: number) => {
  console.log(id);
}

/* ... */

<button
  onClick={() => {
    condition ? onClick(foo) : onClick(bar);
  }}
>
  Do something
</button>
```

**Expected Result**

This rule should not trigger given the code above.

**Actual Result**

```
error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions
```

**Additional Info**

I could find 3 variations of the code above that would pass this rule just fine:

```tsx
<button
  onClick={() => {
    onClick(condition ? foo : bar);
  }}
>
  Do something
</button>

<button onClick={() => onClick(condition ? foo : bar)}>
  Do something
</button>

<button onClick={() => (condition ? onClick(foo) : onClick(bar))}>
  Do something
</button>
```

I am switching to one of those to silence the linter (and because they're better anyway), but I believe the linter should not fail on the input code.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.24.0` |
| `@typescript-eslint/parser`        | `2.24.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.0` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
a ternary is an expression.
If you have a ternary in a statement position, it is an expression whose output is unused, and thus the expression is unused.
If you pass a ternary into a function, or have the ternary as the body of a bodyless arrow function, the return value is used, and thus the expression is used.

This is expected. There is an option if you want to allow unused ternaries in your codebase.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-expressions.md#options

My suggestion is to just use an if statement, but you do you 😄

**Comment by astorije** [comment]
Thanks @bradzacher, that's really great feedback. You are very correct that ternaries are expression, I don't know why I looked over this 😅

It was a trivial enough situation that I could simply do `onClick(condition ? foo : bar)}`, but otherwise an `if` was absolutely the right answer.

Thanks for your time and apologies for wasting your time, I was focusing on the wrong thing. I guess I need to do some rubber duck debugging in those times of confinement lol
